### PR TITLE
[Instrumentation.AWSLambda] Incoming FaaS Span attributes: detect and set attribute for cold start

### PR DESCRIPTION
**Pull request by rypdal**
Fixes #
Design discussion issue #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by Kielek — APPROVED** [review]
LGTM.
Consider if any tests needs to be added/extended.

File: src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs [hunk 1]
```diff
@@ -160,7 +162,14 @@ public static Task<TResult> TraceAsync<TInput, TResult>(
             }
         }
 
-        var functionTags = AWSLambdaUtils.GetFunctionTags(input, context);
+        // No parallel invocation of the same lambda handler.
+        bool faasColdStart = isColdStart;
+        if (faasColdStart)
```
Comment (Oberon00) [inline src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs:167]: The `if` can be removed, it's fine to unconditionally set it to false here. Then we also might not need the local variable if you change it after calculating functionTags

**Comment by Oberon00** [comment]
Yes, this should be unit-testable, please add or extend one to verify the coldstart attribute going from true on the first call to false on subsequent ones.

File: src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs (hunk 1 above)
Comment (rypdal) [reply src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs]: Yes, we can do it. I removed the local variable and moved setting of the internal one after the calculation of functionTags.

**Comment by rypdal** [comment]
> LGTM. Consider if any tests needs to be added/extended.

I added test for the cold start tag check.

**Comment by Kielek** [comment]
One more thing, please extend changelog.

**Comment by rypdal** [comment]
> One more thing, please extend changelog.

changelog has been updated.

File: src/OpenTelemetry.Instrumentation.AWSLambda/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Added detection of Lambda cold start and setting of `faas.coldstart` Activity tag.
```
Comment (Oberon00) [inline src/OpenTelemetry.Instrumentation.AWSLambda/CHANGELOG.md:5]:
```suggestion
* Add detection of Lambda cold start and set `faas.coldstart` Activity tag.
```

**Review by Oberon00 — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

**Pull request merged**